test/vm.py

```diff
@@ -93,33 +94,46 @@ def find_ovmf():
 
 class QEMU(VM):
     MEM = "2000"
-    # TODO: support qemu-system-aarch64 too :)
-    QEMU = "qemu-system-x86_64"
 
-    def __init__(self, img, snapshot=True, cdrom=None):
+    def __init__(self, img, arch="", snapshot=True, cdrom=None):
         super().__init__()
         self._img = pathlib.Path(img)
         self._qmp_socket = self._img.with_suffix(".qemp-socket")
         self._qemu_p = None
         self._snapshot = snapshot
         self._cdrom = cdrom
         self._ssh_port = None
+        if not arch:
+            arch = platform.machine()
+        self._arch = arch
 
     def __del__(self):
         self.force_stop()
 
-    # XXX: move args to init() so that __enter__ can use them?
-    def start(self, wait_event="ssh", snapshot=True, use_ovmf=False):
-        if self.running():
-            return
-        log_path = self._img.with_suffix(".serial-log")
-        self._ssh_port = get_free_port()
-        self._address = "localhost"
-        qemu_cmdline = [
-            self.QEMU, "-enable-kvm",
+    def _gen_qemu_cmdline(self, snapshot, use_ovmf):
+        if self._arch in ("arm64", "aarch64"):
+            qemu_cmdline = [
+                "qemu-system-aarch64",
+                "-machine", "virt",
+                "-cpu", "cortex-a57",
+                "-smp", "2",
+                "-bios", "/usr/share/AAVMF/AAVMF_CODE.fd",
+            ]
+        elif self._arch in ("amd64", "x86_64"):
```

Comment:
i also see the options here, so I'm assuming command line it should only accept amd64 or arm64?

Reply:
We should eventually support all of the platforms that RHEL supports, otherwise our parent company will not be happy.

Reply:
We definitely need to test more scenarios here but if the container is available for both host and target arch  and qemu-user is available for the target arch the existing code should DTRT even for e.g. ppc64le.